pkg/cloudprovider/cloud.go

```diff
@@ -185,4 +185,14 @@ type Zone struct {
 type Zones interface {
 	// GetZone returns the Zone containing the current failure zone and locality region that the program is running in
 	GetZone() (Zone, error)
+
+	// GetZoneByProviderID returns the Zone containing the current zone and locality region of the node specified by providerId
+	// This method is particularly used in the context of external cloud providers where node initialization must be down
+	// outside the kubelets.
+	GetZoneByProviderID(providerID string) (Zone, error)
+
+	// GetZoneByNodeName returns the Zone containing the current zone and locality region of the node specified by node name
+	// This method is particularly used in the context of external cloud providers where node initialization must be down
+	// outside the kubelets.
+	GetZoneByNodeName(nodeName types.NodeName) (Zone, error)
```

Comment:
Not a huge fan of having both `GetZoneByProviderID` and `GetZoneByNodeName`, but this is following a pattern in other cloud provider interfaces. Aside from having to maintain/support the methods, it does give external cloud providers flexibility in terms of how they would like to fetch node data (by id or name).